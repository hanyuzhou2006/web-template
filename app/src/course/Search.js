import React, { Component } from 'react';


export default class Search extends Component {

    render() {
        const { value = "", onSearch } = this.props;
        const className = `btn clear-btn ${value === '' ? 'hidden' : ''}`
        return (
            <div className="search-container">
                <div className="search">
                    <input placeholder="搜索" value={value} onChange={onSearch} />
                    <button className={className} onClick={() => { onSearch({ target: { value: "" } }) }}>
                        <i className="mi mi-Clear"></i>
                    </button>
                </div>
            </div>
        )
    }
}