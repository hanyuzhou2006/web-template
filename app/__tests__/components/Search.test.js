import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Search from '../../src/course/Search';



test('render normal', () => {
    const component = renderer.create(
        <Search value='hello' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('render empty', () => {
    const component = renderer.create(
        <Search value='' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('vdom render normal', () => {
    const search = mount(
        <Search value='hello' />
    );
    const input = search.find('input');
    expect(input.prop('value')).toBe('hello');

    const button = search.find('button.hidden');
    expect(button.length).toBe(0);

})

test('vdom render when value is empty', () => {
    const search = mount(
        <Search value='' />
    );
    const button = search.find('button.hidden');
    expect(button.length).toBe(1);
})




