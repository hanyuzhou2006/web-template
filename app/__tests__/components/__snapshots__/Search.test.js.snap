exports[`test render empty 1`] = `
<div
  className="search-container">
  <div
    className="search">
    <input
      onChange={undefined}
      placeholder="搜索"
      value="" />
    <button
      className="btn clear-btn hidden"
      onClick={[Function]}>
      <i
        className="mi mi-Clear" />
    </button>
  </div>
</div>
`;

exports[`test render normal 1`] = `
<div
  className="search-container">
  <div
    className="search">
    <input
      onChange={undefined}
      placeholder="搜索"
      value="hello" />
    <button
      className="btn clear-btn "
      onClick={[Function]}>
      <i
        className="mi mi-Clear" />
    </button>
  </div>
</div>
`;
